agent/engine/task_manager.go

```diff
@@ -765,8 +765,9 @@ func (mtask *managedTask) progressTask() {
 	changed := mtask.UpdateStatus()
 	if changed {
 		seelog.Debugf("Managed task [%s]: container or resource change also resulted in task change", mtask.Arn)
+
 		// If knownStatus changed, let it be known
-		mtask.emitTaskEvent(mtask.Task, "")
+		mtask.emitTaskEvent(mtask.Task, mtask.Task.GetTerminalReason())
```

Comment:
Task's status change does not necessarily indicate terminal state. It would be good to check if the known status is `STOPPED` and if so, emit event with `terminalReason`.

Reply:
good catch.